asapdiscovery-modeling/asapdiscovery/modeling/protein_prep_v2.py

```diff
@@ -0,0 +1,123 @@
+import abc
+from pathlib import Path
+from typing import Literal, Optional
+
+import yaml
+from asapdiscovery.data.openeye import oechem
+from asapdiscovery.data.utils import seqres_to_res_list
+from asapdiscovery.modeling.modeling import (
+    make_design_unit,
+    mutate_residues,
+    spruce_protein,
+    superpose_molecule,
+)
+from pydantic import BaseModel, Field, root_validator
+
+
+class ProteinPrepperBase(BaseModel):
```

Comment:
do we need a base class?

Reply:
Future prep, I suppose its not strictly necessary, but perhaps we will have other ways in the future. Its more for typing in a large workflow down the track.